x/incentive/keeper/keeper.go

```diff
@@ -39,10 +45,267 @@ func NewKeeper(
 		storeKey:   storeKey,
 		memKey:     memKey,
 		paramstore: ps,
-		commitmentKeeper: ck,
+		cmk:        ck,
+		stk:        sk,
+		tci:        &types.TotalCommitmentInfo{},
 	}
 }
 
 func (k Keeper) Logger(ctx sdk.Context) log.Logger {
 	return ctx.Logger().With("module", fmt.Sprintf("x/%s", types.ModuleName))
 }
+
+// Update total commitment info
+func (k Keeper) UpdateTotalCommitmentInfo(ctx sdk.Context, epochIdentifier string) {
+	// Fetch total staked Elys amount again
+	k.tci.TotalElysBonded = k.stk.TotalBondedTokens(ctx)
+	// Initialize with amount zero
+	k.tci.TotalEdenCommitted = sdk.ZeroInt()
+	k.tci.TotalEdenBoostCommitted = sdk.ZeroInt()
+
+	// Iterate to calculate total Eden and Eden boost committed
+	k.cmk.IterateCommitments(ctx, epochIdentifier, func(commitments ctypes.Commitments) bool {
+		committedTokens := commitments.CommittedTokens
+		// Sum Eden and Eden boost committed
+		for _, c := range committedTokens {
+			// Eden
+			if c.Denom == types.Eden {
+				k.tci.TotalEdenCommitted = k.tci.TotalEdenCommitted.Add(c.Amount)
+			}
+
+			// Eden boost
+			if c.Denom == types.EdenB {
+				k.tci.TotalEdenBoostCommitted = k.tci.TotalEdenCommitted.Add(c.Amount)
+			}
+		}
+
+		return false
+	})
+}
```

Comment:
@kenta-elys We may be able to improve this by tracking the totalcommitted amounts for each denom in the commitment module and updating them when changes are made. That way we don't have to iterate all commitments every epoch.

Reply:
Got it.